packages/manager/apps/pci-object-storage/src/pages/objects/container/users/Listing.page.tsx

```diff
@@ -1,3 +1,254 @@
+import { Suspense, useRef, useState } from 'react';
+import {
+  OsdsButton,
+  OsdsIcon,
+  OsdsMessage,
+  OsdsPopover,
+  OsdsPopoverContent,
+  OsdsSearchBar,
+  OsdsSpinner,
+  OsdsText,
+} from '@ovhcloud/ods-components/react';
+import {
+  ODS_BUTTON_SIZE,
+  ODS_BUTTON_VARIANT,
+  ODS_ICON_NAME,
+  ODS_ICON_SIZE,
+  ODS_MESSAGE_TYPE,
+  ODS_SPINNER_SIZE,
+} from '@ovhcloud/ods-components';
+import {
+  ODS_THEME_COLOR_INTENT,
+  ODS_THEME_TYPOGRAPHY_LEVEL,
+  ODS_THEME_TYPOGRAPHY_SIZE,
+} from '@ovhcloud/ods-common-theming';
+import { FilterCategories } from '@ovh-ux/manager-core-api';
+import { useTranslation } from 'react-i18next';
+import { Outlet, useNavigate, useParams } from 'react-router-dom';
+import { usePaginatedUsers } from '@/api/hooks/useUser';
+import { useDatagridColumn } from './useDatagridColumn';
+import { PCI_FEATURES_FREE_LOCAL_ZONES_BANNER } from '@/constants';
+import queryClient from '@/queryClient';
+
 export default function Listing() {
-  return <div>User listing</div>;
+  const { t } = useTranslation('objects/users');
+  const { t: tFilter } = useTranslation('filters');
+
+  const { projectId } = useParams();
+  const { pagination, setPagination, sorting, setSorting } = useDataGrid();
+
+  const navigate = useNavigate();
+  const { clearNotifications } = useNotifications();
+  const { filters, addFilter, removeFilter } = useColumnFilters();
+  const [searchField, setSearchField] = useState('');
+  const [searchQueries, setSearchQueries] = useState<string[]>([]);
+  const filterPopoverRef = useRef(undefined);
+
+  const { allUsers, paginatedUsers, isPending } = usePaginatedUsers(
+    projectId,
+    pagination,
+    sorting,
+    filters,
+  );
+
+  const columns = useDatagridColumn();
+
+  const { data: availability } = useFeatureAvailability([
+    PCI_FEATURES_FREE_LOCAL_ZONES_BANNER,
+  ]);
+
+  const refresh = async () => {
+    queryClient.invalidateQueries({
+      queryKey: ['project', projectId, 'users'],
+    });
+    queryClient.invalidateQueries({
+      queryKey: ['user'],
+    });
+  };
+
+  return (
+    <RedirectionGuard
+      isLoading={isPending}
+      condition={!isPending && allUsers?.length === 0}
+      route="../onboarding"
+    >
+      <div className="header mt-8">
+        <Notifications />
+      </div>
+      <OsdsText
+        level={ODS_THEME_TYPOGRAPHY_LEVEL.heading}
+        color={ODS_THEME_COLOR_INTENT.text}
+        size={ODS_THEME_TYPOGRAPHY_SIZE._400}
+        className="mt-6 block"
+      >
+        {t('pci_projects_project_storages_containers_users_title')}
+      </OsdsText>
+      <OsdsText
+        level={ODS_THEME_TYPOGRAPHY_LEVEL.body}
+        color={ODS_THEME_COLOR_INTENT.text}
+        size={ODS_THEME_TYPOGRAPHY_SIZE._400}
+        className="mt-6 block"
+      >
+        {t('pci_projects_project_storages_containers_users_user_description')}
+      </OsdsText>
+      {availability && availability[PCI_FEATURES_FREE_LOCAL_ZONES_BANNER] && (
```

Comment:
To prevent Sonar from flagging it, you can rewrite:
```
availability && availability[PCI_FEATURES_FREE_LOCAL_ZONES_BANNER] as
availability?.[PCI_FEATURES_FREE_LOCAL_ZONES_BANNER]."
```